pkg/btf/btf.go

```diff
@@ -101,3 +104,118 @@ func InitCachedBTF(lib, btf string) error {
 func GetCachedBTFFile() string {
 	return btfFile
 }
+
+func resolveNestedTypedef(ty btf.Type) btf.Type {
+	if t, ok := ty.(*btf.Typedef); ok {
+		return resolveNestedTypedef(t.Type)
+	}
+	return ty
+}
+
+/*
+ResolveBtfPath function recursively search in a btf structure in order to
+found a specific path until it reach the target or fail.
+
+The function also search in embedded anonymous structures or unions to cover as
+much use cases as possible. For instance, mm_struct have 2 fields, anonymous
+struct and another type. But you are still able to look into the anonymous
+struct by specifying a path like "mm.pgd.pgd".
+
+@btfArgs: dest array for storing btf informations to reach the target on the
+bpf side.
+@currentType: The current type being proccessed, starts with root type.
+@pathToFound: The string representation of the path to reach in the structures.
+@i: The current depth, until last element of pathToFound.
+
+Return: The last type found matching the path, or error.
+*/
+func ResolveBtfPath(
+	btfArgs *[api.MaxBtfArgDepth]api.ConfigBtfArg,
+	currentType btf.Type,
+	pathToFound []string,
+	i int,
+) (*btf.Type, error) {
+	switch t := currentType.(type) {
+	case *btf.Struct:
+		return processMembers(btfArgs, currentType, t.Members, pathToFound, i)
+	case *btf.Union:
+		return processMembers(btfArgs, currentType, t.Members, pathToFound, i)
+	case *btf.Pointer:
+		if len(pathToFound[i]) == 0 {
+			(*btfArgs)[i].IsPointer = uint16(1)
+			(*btfArgs)[i].IsInitialized = uint16(1)
+			return ResolveBtfPath(btfArgs, t.Target, pathToFound, i+1)
+		}
+		(*btfArgs)[i-1].IsPointer = uint16(1)
```

Comment:
Why is the above line needed?
As I commented in the bpf code, it's not clear to me what it means to have `IsPonter` equal to `false` but `IsIntialized` set to `true`.

Reply:
I see. Thanks!

I ran a few examples, and I think I understand it better now.
One thing I was not able to figure out is when the "len(pathToFound[i]) == 0" condition will trigger.
Can you provide an example?

Reply:
Yes, this condition mostly matches the "padding" system. This is the way I found to handle nested pointer such as `struct cred** new` https://elixir.bootlin.com/linux/v6.13-rc3/source/security/security.c#L2751

So in case of nested pointers, the path will look like this
["",..,"","my","path"]

So since current path is "", this means that the pointer is not the previous type but the current. So I add this specific use case.